parser/parser.go

```diff
@@ -162,6 +160,87 @@ func (p *parser) parseCommaList(end tokenKind, elementKind string) (*token, ast.
 	}
 }
 
+func (p *parser) parseArgument() (*ast.Identifier, ast.Node, error) {
+	var id *ast.Identifier
+	if p.peek().kind == tokenIdentifier && p.doublePeek().kind == tokenOperator && p.doublePeek().data == "=" {
+		ident := p.pop()
+		var tmpID = ast.Identifier(ident.data)
+		id = &tmpID
+		p.pop() // "=" token
+	}
+	expr, err := p.parse(maxPrecedence)
+	if err != nil {
+		return nil, nil, err
+	}
+	return id, expr, nil
+}
+
+// TODO(sbarzowski) - this returned bool is weird
+// TODO(sbarzowski) - name - it's also used for parameters
+func (p *parser) parseArguments(end tokenKind, elementKind string) (*token, *ast.Arguments, bool, error) {
+	args := &ast.Arguments{}
+	gotComma := false
+	namedArgumentAdded := false
+	first := true
+	for {
+		next := p.peek()
+		if !first && !gotComma {
+			if next.kind == tokenComma {
+				p.pop()
+				next = p.peek()
+				gotComma = true
+			}
+		}
+
+		// TODO(check specifically for eof)
+
+		if next.kind == end {
+			// gotComma can be true or false here.
+			return p.pop(), args, gotComma, nil
+		}
+
+		if !first && !gotComma {
+			return nil, nil, false, MakeStaticError(fmt.Sprintf("Expected a comma before next %s, got %s.", elementKind, next), next.loc)
+		}
+
+		id, expr, err := p.parseArgument()
+		if err != nil {
+			return nil, nil, false, err
+		}
+		if id == nil {
+			if namedArgumentAdded {
+				return nil, nil, false, MakeStaticError("Positional argument after a named argument is not allowed", next.loc)
+			}
+			args.Positional = append(args.Positional, expr)
+		} else {
+			namedArgumentAdded = true
+			args.Named = append(args.Named, ast.NamedArgument{Name: *id, Arg: expr})
+		}
+		gotComma = false
+		first = false
+	}
+}
+
+// TODO(sbarzowski) - this returned bool is weird
+func (p *parser) parseParameters(end tokenKind, elementKind string) (*token, *ast.Parameters, bool, error) {
```

Comment:
Why return *token when it seems to get ignored by each invocation?

Reply:
parseParameters consumes parameters and the closing paren. This closing paren is potentially interesting for its location (it is used in parseArguments). 

For now I just removed it. 

Perhaps it shouldn't consume the closing paren (it can just peek). Opening paren is consumed by the caller, so perhaps it would be more expected. Or perhaps it should consume both? Maybe the location could be saved in ast.Arguments and we would avoid the weirdness of returning the token from parseArguments.